src/twister2/log_parser/ztest_log_parser.py

```diff
@@ -29,15 +34,16 @@ class ZtestLogParser(LogParserAbstract):
     def __init__(self, stream: Iterator[str], *, ignore_faults: bool = False, **kwargs):
         super().__init__(stream, **kwargs)
         self.detected_suite_names: list[str] = []
+        self.subtest_results: list[SubTestResult] = []
         self.ignore_faults = ignore_faults
 
-    def parse(self, timeout: float = 60) -> Generator[SubTestResult, None, None]:
-        """Parse logs and return list of tests with statuses."""
+    def parse(self, timeout: float = 60, pytest_subtests: SubTests | None = None) -> None:
```

Comment:
I don't see any reason to pass whole module to `parse` method. Can we just use simple `bool` flag instead?

Reply:
No, I don't pass whole module. I only pass object of `SubTests` class, which was created during calling fixture `subtests` in test definition:
https://github.com/zephyrproject-rtos/twister/blob/a038baeaa592b5e2b545b06fbf00b51a908200df/src/twister2/yaml_test_function.py#L58-L65
https://github.com/zephyrproject-rtos/twister/blob/a038baeaa592b5e2b545b06fbf00b51a908200df/src/twister2/yaml_test_function.py#L73

The simplest way to achieve this `SubTest` object is call `subtests` fixture - it is hard to do it in other place - for example in  `ztest_log_parser.py`. To create `SubTest` object you need to have access to `FixtureRequest`: 
https://github.com/pytest-dev/pytest-subtests/blob/main/pytest_subtests.py#L97-L111

I agree that name which I proposed in this PR for `SubTest` object is misleading (`pytest_subtests`) and I can propose better and clearer name.

Reply:
Ok. I just missed this part of code where subtests in passed to the yaml test function as a fixture. 
Maybe better name would be `subtests_fixture`?